lib/api.c

```diff
@@ -170,6 +170,17 @@ static struct pqos_api {
                        const unsigned num_cos,
                        const struct pqos_mba *requested,
                        struct pqos_mba *actual);
+        /** Get SMBA */
+        int (*smba_get)(const unsigned smba_id,
+                        const unsigned max_num_cos,
+                        unsigned *num_cos,
+                        struct pqos_mba *smba_tab);
+        /** Set SMBA mask */
+        int (*smba_set)(const unsigned smba_id,
```

Comment:
Please avoid adding these 2 new APIs. Use existing APIs mba_get and mba_set to call hw_smba_get_amd, hw_smba_set_amd, os_smba_get_amd, os_smba_set_amd functions. You can use structure pqos_mba ctrl variable to differentiate the mba and smba.

Reply:
This gets a little bit complicated.
The APIs mba_get and mba_set are initialized statically during the init.
api.mba_get = hw_mba_get_amd;
api.mba_set = hw_mba_set_amd;
....
api.mba_get = os_mba_get_amd;
api.mba_set = os_mba_set_amd;

With your suggested comment, the function hw_mba_get_amd would need to handle functionalities for both mba and smba. Basically inside the function hw_mba_get_amd, I need to call hw_smba_get_amd if pqos_mba.ctrl is set to smba. Also, need to add a new definition for pqos_mba.ctrl variable. The ctrl variable has been used all over the code.
Same with hw_mba_set_amd, os_mba_get_amd and os_mba_set_amd.

This can be done. But it will be a bit messy to maintain I feel. These are two different features, so I kept them separate, and pretty easy-to-spot issues. What do you think?

Reply:
Need to keep less APIs. Declare new variable in pqos_mba structure to differentiate the mba & smba.